### ISSUE TITLE
Blacklisted tokens can still be refreshed

### ISSUE THREAD

**Issue by ashokdelphia** [Styria-Digital/django-rest-framework-jwt#36]
If you configure an app to allow blacklisting tokens and turn `JWT_ALLOW_REFRESH` on, then you can still operate the refresh endpoint using a blacklisted token, which allows you to side-step the fact that the token is invalidated.

**Comment by ashokdelphia** [comment]
There's a more nuanced problem where invalidating a token doesn't do anything to any related tokens that could still be valid.

For example, if you authenticate to create token A, then refresh it to get token B, but A is still valid, then blacklisting token B only prevents that exact token (reasonably enough). It might be worth adding a way to blacklist the whole family of refreshed tokens, by adding a common identifier for the initial authentication, but that's well beyond the scope of what I'm seeking to fix here.

Alternatively, if we were to add something that prevented a token being refreshed more than once (which I would see as a reasonable restriction, although possibly annoying to enforce), then we could potentially use the same mechanism to prevent the 'antecedents' of refreshed tokens from being used, too.

**Comment by fitodic** [comment]
We did set out to implement JWT blacklisting [as described by Auth0](https://auth0.com/blog/blacklist-json-web-token-api-keys/), including the [refresh tokens upon usage](https://auth0.com/blog/refresh-tokens-what-are-they-and-when-to-use-them/), but we ran out of time and resources due to other issues. I've added the links above so anyone who reads this issue knows what was the original intention and based on you comment, figures out what's left to do.

I agree with everything you've pointed out and would like to see this change land, either by reviewing it or doing it myself, but we are a bit short on time at the moment, hence the basic blacklist functionality.
